packages/core/src/baseclient.ts

```diff
@@ -791,11 +794,40 @@ function _validateBeforeSendResult(
         return event;
       },
       e => {
-        throw new SentryError(`\`${beforeSendProcessorName}\` rejected with ${e}`);
+        throw new SentryError(`${beforeSendLabel} rejected with ${e}`);
       },
     );
   } else if (!isPlainObject(beforeSendResult) && beforeSendResult !== null) {
     throw new SentryError(invalidValueError);
   }
   return beforeSendResult;
 }
+
+/**
+ * Process the matching `beforeSendXXX` callback.
+ */
+function processBeforeSend<T extends Event>(
+  options: ClientOptions,
+  event: T,
+  hint: EventHint,
+): PromiseLike<T | null> | T | null {
+  const { beforeSend, beforeSendTransaction } = options;
+
+  if (isErrorEvent(event) && beforeSend) {
+    return beforeSend(event, hint) as PromiseLike<T | null> | T | null;
+  }
+
+  if (isTransactionEvent(event) && beforeSendTransaction) {
+    return beforeSendTransaction(event, hint) as PromiseLike<T | null> | T | null;
+  }
+
+  return event;
+}
+
+function isErrorEvent(event: Event): event is ErrorEvent {
+  return event.type === undefined;
+}
+
+function isTransactionEvent(event: Event): event is TransactionEvent {
+  return event.type === 'transaction';
+}
```

Comment:
Random thought: We have this `is.ts` file in utills. WDYT, does it make sense to move them there?

Reply:
hmm, not sure, that could be confusing, as there it is a lot about built ins etc. there is actually also this in there:

```ts
export function isErrorEvent(wat: unknown): boolean {
  return isBuiltin(wat, 'ErrorEvent');
}
```

but that checks something else I guess 😅